compiler_gym/service/proto/py_converters.py

```diff
@@ -90,6 +90,25 @@ def __call__(self, val: Any) -> Any:
         return self.conversion_map[type(val)](val)
 
 
+class TypeIdDispatchConverter:
```

Comment:
Needs a docstring, with a usage example of how to instantiate it

Reply:
I added as an example the existing unit test.
Ultimately, there should be a short guide on how to customize the conversion from/to proto.
I think the current method here and other converter classes is not good enough. They use dispatching based on dict keys, but this is not flexible enough when you have to convert for example any `numbers.Integral`. After an attempt on conversion with an exact type match it should go through a list of fallback converters that attempt the conversion one-by-one and if none succeeds there should be a failure.

Reply:
SGTM. Could you file a documentation issue requesting this?

Reply:
I opened one at #649.